tests/test_grid.py

```diff
@@ -169,6 +169,36 @@ def test_agent_remove(self):
         assert agent.pos is None
         assert self.grid.grid[x][y] is None
 
+    def test_swap_pos(self):
+
+        # Swap agents positions
+        agent_a, agent_b = list(filter(None, self.grid))[:2]
```

Comment:
Note: the random sampling failed in the first because when the test is moved to `TestBaseGrid`, the `self.grid` in `TestBaseGrid` is not full.